### DISCUSSION TITLE
Allow GitHub App installation token

### DISCUSSION THREAD

**Discussion by Nef10** [BetaHuhn/repo-file-sync-action#89] [Ideas] [answered]
GitHub Apps provide better security than personal access tokens, so it would be cool to have an option to allow them.

I noticed that someone already made some hack #77 fo some basic support (there is also a kind of working version [here](https://github.com/BetaHuhn/repo-file-sync-action/compare/master...Roblox:353a671418479d95f90bb5dc82ab52139300e2bb)). 

As I would like to use it I am willing to implement a proper version. However, before I start forking and doing something, I wanted to check what kind of design you would prefer? Maybe instead of using the `GH_PAT` parameter and the `IS_INSTALLATION_TOKEN` flag, we have a separate `GH_INSTALLATION_TOKEN` parameter which can be provided instead of `GH_PAT`?

I also noticed that `this.github.users.getAuthenticated()` is failing with this kind of token, so we should check that email / username are provided.

**Comment by BetaHuhn** [comment, marked as answer]
Hey, thanks for taking the time to open this discussion!

I would prefer your idea with a separate `GH_INSTALLATION_TOKEN `. It doesn't make any sense to reuse the `GH_PAT` parameter for anything other than providing a personal access token.

Would be great if you create a PR for this!

**Reply by BetaHuhn** [reply]
The `GH_INSTALLATION_TOKEN` parameter is included in [v1.11.0](https://github.com/BetaHuhn/repo-file-sync-action/releases/tag/v1.11.0) 🎉

**Reply by Nef10** [reply]
Hey, unfortunately it seems like your release bot did not actually built the software, as the file in `dist` is unchanged and thus the feature is not included.

**Reply by BetaHuhn** [reply]
Whoops, thanks for reporting! Should be fixed now!
